packages/spectral/src/types/ComponentRegistry.ts

```diff
@@ -0,0 +1,140 @@
+import { ComponentManifest, PermissionAndVisibilityType } from ".";
+import { UnionToIntersection } from "./utils";
+
+/**
+ * Root ComponentRegistry type exposed for augmentation.
+ *
+ * The expected interface when augmenting is:
+ *
+ * ```ts
+ * interface IntegrationDefinitionComponentRegistry {
+ *   [key: string]: ComponentManifest
+ * }
+ * ```
+ *
+ */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface IntegrationDefinitionComponentRegistry {}
+
+export type ComponentRegistry =
+  keyof IntegrationDefinitionComponentRegistry extends never
+    ? {
+        [key: string]: ComponentManifest;
+      }
+    : UnionToIntersection<
+        keyof IntegrationDefinitionComponentRegistry extends infer TComponentKey
+          ? TComponentKey extends keyof IntegrationDefinitionComponentRegistry
+            ? {
+                [Key in TComponentKey]: IntegrationDefinitionComponentRegistry[TComponentKey];
+              }
+            : never
+          : never
+      >;
+
+interface ConnectionInputPermissionAndVisibility {
+  /**
+   * Optional value that sets the permission and visibility of the Config Var. @default "customer"
+   *
+   * "customer" - Customers can view and edit the Config Var.
+   * "embedded" - Customers cannot view or update the Config Var as the value will be set programmatically.
+   * "organization" - Customers cannot view or update the Config Var as it will always have a default value or be set by the organization.
+   */
+  permissionAndVisibilityType?: PermissionAndVisibilityType;
+  /** Optional value that specifies whether this Config Var is visible to an Organization deployer. @default true */
+  visibleToOrgDeployer?: boolean;
```

Comment:
We have two interfaces that re-declare both these types, wonder if we should Pick or extend one.